modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/facade/XSKHDBCoreFacade.java

```diff
@@ -536,4 +552,21 @@ private String getContent(IResource resource) throws SynchronizationException {
     }
     return contentAsString;
   }
+
+  private String getType(String resourceExtension) {
+    return parserTypes.get(resourceExtension);
+  }
+
+  private boolean isParsed(String location, String content, String resourceExtension)
+      throws XSKDataStructuresException, ClassNotFoundException, InstantiationException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+
+    String modelType = getType(resourceExtension);
+    Class<XSKDataStructureModel> clazz = xskCoreParserService.getDataStructureClass(modelType);
+    XSKDataStructureModel baseDataStructureModel = clazz.getDeclaredConstructor().newInstance();
+    baseDataStructureModel.setLocation(location);
+    baseDataStructureModel.setType(modelType);
+    baseDataStructureModel.setHash(DigestUtils.md5Hex(content));
+
+    return managerServices.get(baseDataStructureModel.getType()).existsArtifactMetadata(baseDataStructureModel);
```

Comment:
There should be check for **null**, so that if we introduce support for new file extension and forgot to add it to the list of known file extensions, the **isParsed()** method will fail with **NPE**. In case the **modelType** is **null**, then the **isParsed()** method should return **false**.

Reply:
Done in ab8c929